compiler/rustc_mir_build/src/builder/scope.rs

```diff
@@ -1552,7 +1552,7 @@ impl<'a, 'tcx: 'a> Builder<'a, 'tcx> {
             let mut unwind_indices = IndexVec::from_elem_n(unwind_target, 1);
             for (drop_idx, drop_node) in drops.drops.iter_enumerated().skip(1) {
                 match drop_node.data.kind {
-                    DropKind::Storage => {
+                    DropKind::Storage | DropKind::ForLint => {
```

Comment:
This is the fix for the ICE. Not totally sure why it fixes it :?

Reply:
Hmm. I think it fixes it because we generally copied the "storage" logic -- i.e., we only adjust `unwind_to` when `storage_dead_on_unwind` is true. I'm not 100% sure what the "add entry point" business is about though.

Reply:
OK, I did a bit of looking, I think this is correct, and it has to do with the fact that we are inserting a statement into a block and not a terminator.